cogniac/gateway.py

```diff
@@ -35,23 +50,88 @@ class CogniacGateway(object):
     media from another host on the same network.
     """
 
-    def __init__(self, timeout=60, url_prefix=None):
+    ##
+    #  get
+    ##
+    @classmethod
+    @retry(stop_max_attempt_number=8, wait_exponential_multiplier=500, retry_on_exception=server_error)
+    def get(cls,
+            connection=None,
+            gateway_id=None,
+            url_prefix=None):
+        """
+        get single network camera
+        connnection (CogniacConnection): Authenticated CogniacConnection object
+        gateway_id (String): the unique identifier of a Gateway object
+        returns CogniacGateway object
+        """
+        gateway_dict = None
+        if connection and gateway_id:
+            resp = connection._get("/gateways/{}".format(gateway_id))
+            gateway = resp.json()
+        return CogniacGateway(connection=connection, url_prefix=url_prefix, gateway_dict=gateway)
```

Comment:
`gateway` undefined if conditional not meet.  Line 68 should be `gateway = None`, right?

Reply:
I'm thinking that it should be possible to directly instantiate the `CogniacGateway` to connect to an EdgeFlow without first authenticating with a Core.

The idea here was that `gateway_id` can be `None` and `CogniacGateway` can be instantiated with only `url_prefix` for the EF's local API. This seems useful for cases in which a user wants to use the SDK to upload images to an EdgeFlow. Note, in this case, the Core APIs wouldn't be available, so first creating a `CogniacConnection` object seems unnecessary.

I'm trying to push validation logic into the `CogniacGateway` initialize to ensure that the `CogniacGateway` is valid for requests against (1) local EF API only, (2) CloudCore APIs only, (3) both.

Does this seem reasonable?

FWIW, I think the committed code handles these cases, but it may not be clear without some examples. I will add those to the PR after testing.